cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/transform/ValidatorTransform.java

```diff
@@ -0,0 +1,159 @@
+    "      } else if (!apacheValidator.isInRange(input.content_length, 0, 1024 * 1024)) {" +
+    "         isValid = false; errMsg = \"content length >1MB\"; errCode = 10;" +
+    "      }" +
+    "      resultMap.put(\"isValid\", isValid.toString()); " +
+    "      resultMap.put(\"errorCode\", errCode); " +
+    "      resultMap.put(\"errorMsg\", errMsg); " +
+    "      return resultMap;" +
+    "   };" +
+    "This Javascript function in example uses apache validation functions";
+
+  private static final Gson GSON = new GsonBuilder()
+    .registerTypeAdapter(StructuredRecord.class, new StructuredRecordSerializer())
+    .create();
+  private static final Logger LOG = LoggerFactory.getLogger(ValidatorTransform.class);
+
+  private ValidatorConfig config;
+  private Metrics metrics;
+  private Invocable invocable;
+
+  // for unit tests, otherwise config is injected by plugin framework.
+  public ValidatorTransform(ValidatorConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
+    for (String validatorName : config.validators.split(",")) {
+      // the validtor's have only type and name, passing the properties and id as null.
+      pipelineConfigurer.usePluginClass("validator", validatorName, validatorName,
+                                        PluginProperties.builder().build());
+    }
+  }
+
+  @Override
+  public void initialize(TransformContext context) throws Exception {
+    List<Validator> validators = new ArrayList<>();
+    for (String pluginId : config.validators.split(",")) {
+      validators.add((Validator) context.newPluginInstance(pluginId));
+    }
+    try {
+      setUpInitialScript(context, validators);
+    }  catch (ScriptException e) {
+      throw new IllegalArgumentException("Invalid script.", e);
+    }
+  }
+
+  @VisibleForTesting
+  void setUpInitialScript(TransformContext context, List<Validator> validators) throws ScriptException {
+    ScriptEngine engine;
+    ScriptEngineManager manager = new ScriptEngineManager();
+    engine = manager.getEngineByName("JavaScript");
+    String scriptStr = config.validationScript;
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(scriptStr), "Filter script must be specified.");
+
+    for (Validator validator : validators) {
+      engine.put(validator.getValidatorVariableName(), validator.getValidator());
+    }
+    engine.eval(scriptStr);
+    invocable = (Invocable) engine;
+    metrics = context.getMetrics();
+  }
+
+  @Override
+  public void transform(StructuredRecord input, Emitter<StructuredRecord> emitter) throws Exception {
+    try {
+      Map<String, String> result = (Map<String, String>) invocable.invokeFunction("isValid", GSON.toJson(input));
+      if (Boolean.parseBoolean(result.get("isValid"))) {
+        emitter.emit(input);
+      } else {
+        emitter.emitError(input);
```

Comment:
this is where we need some `InvalidEntry<StructuredRecord>` object to avoid dropping code and message.